docusaurus/docs/iOS/controllers/channels.md

```diff
@@ -0,0 +1,204 @@
+        controller = ChatClient.shared.channelController(for: cid)
+        controller.synchronize()
+
+        controller.delegate = typingIndicatorView
+        navigationItem.titleView = typingIndicatorView
+    }
+}
+```
+
+## ChannelMemberListController
+
+`ChatChannelMemberListController` allows you to observe and mutate data and observing changes for a list of channel members based on the provided query.
+
+### ChannelMemberList Delegate
+
+Classes that conform to the `ChatChannelMemberListControllerDelegate` protocol will data and changes for a list of members queried by the controller.
+
+```swift
+func memberListController(
+    _ controller: ChatChannelMemberListController,
+    didChangeMembers changes: [ListChange<ChatChannelMember>]
+)
+```
+
+### Example: Listing Members
+
+This example uses the `ChatChannelMemberListController` to fetch all members on the channel `messaging:123` and sets a view as its delegate.
+
+```swift
+class MembersSearchResultsView: UIView, ChatChannelMemberListControllerDelegate {
+    var labelView: UILabel = {
+        var label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+
+    override public init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        addSubview(labelView)
+
+        NSLayoutConstraint.activate([
+            labelView.topAnchor.constraint(equalTo: topAnchor),
+            labelView.bottomAnchor.constraint(equalTo: bottomAnchor),
+            labelView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            labelView.trailingAnchor.constraint(equalTo: trailingAnchor)
+        ])
+    }
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
+
+    func memberListController(_ controller: ChatChannelMemberListController, didChangeMembers changes: [ListChange<ChatChannelMember>]) {
+        let membersStr = controller.members.map { $0.name ?? $0.id }.joined(separator: ", ")
+        labelView.text = membersStr
+
+        print("\(changes.count) members changed (added/inserted/deleted)")
+    }
+}
+
+class ViewController: UIViewController {
+    var controller: ChatChannelMemberListController!
+
+    var membersSearchResultsView: MembersSearchResultsView = {
+        var view = MembersSearchResultsView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        let cid = ChannelId.init(type:.messaging, id:"123")
+        let query = ChannelMemberListQuery.init(cid: cid)
+        
+        controller = ChatClient.shared.memberListController(query: query)
+        controller.delegate = membersSearchResultsView
+
+        controller.synchronize {error in
+            if error != nil {
+                log.assertionFailure(error!)
+            }
+        }
+
+        navigationItem.titleView = membersSearchResultsView
+    }
+}
+```
```

Comment:
Consider breaking this block into smaller blocks with prefacing content explaining the "Why" behind each of the steps or adding some comments above the two classes explaining what each does at a high level.